src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -804,60 +739,49 @@ public override Task WriteAsync(ReadOnlyMemory<char> buffer, CancellationToken c
         // to ensure performant access inside the state machine that corresponds this async method.
         // Fields that are written to must be assigned at the end of the method *and* before instance invocations.
-        private static async Task WriteAsyncInternal(StreamWriter _this, ReadOnlyMemory<char> source,
-                                                     char[] charBuffer, int charPos, int charLen, char[] coreNewLine,
-                                                     bool autoFlush, bool appendNewLine, CancellationToken cancellationToken)
+        private async Task WriteAsyncInternal(ReadOnlyMemory<char> source, bool appendNewLine, CancellationToken cancellationToken)
         {
             int copied = 0;
             while (copied < source.Length)
             {
-                if (charPos == charLen)
+                if (_charPos == _charLen)
                 {
-                    await _this.FlushAsyncInternal(false, false, charBuffer, charPos, cancellationToken).ConfigureAwait(false);
-                    Debug.Assert(_this._charPos == 0);
-                    charPos = 0;
+                    await FlushAsyncInternal(flushStream: false, flushEncoder: false, cancellationToken).ConfigureAwait(false);
                 }
 
-                int n = Math.Min(charLen - charPos, source.Length - copied);
+                int n = Math.Min(_charLen - _charPos, source.Length - copied);
                 Debug.Assert(n > 0, "StreamWriter::Write(char[], int, int) isn't making progress!  This is most likely a race condition in user code.");
 
-                source.Span.Slice(copied, n).CopyTo(new Span<char>(charBuffer, charPos, n));
-                charPos += n;
+                source.Span.Slice(copied, n).CopyTo(new Span<char>(_charBuffer, _charPos, n));
+                _charPos += n;
                 copied += n;
             }
 
             if (appendNewLine)
             {
-                for (int i = 0; i < coreNewLine.Length; i++)   // Expect 2 iterations, no point calling BlockCopy
+                for (int i = 0; i < CoreNewLine.Length; i++) // Generally 1 (\n) or 2 (\r\n) iterations
                 {
-                    if (charPos == charLen)
+                    if (_charPos == _charLen)
                     {
-                        await _this.FlushAsyncInternal(false, false, charBuffer, charPos, cancellationToken).ConfigureAwait(false);
-                        Debug.Assert(_this._charPos == 0);
-                        charPos = 0;
+                        await FlushAsyncInternal(flushStream: false, flushEncoder: false, cancellationToken).ConfigureAwait(false);
                     }
 
-                    charBuffer[charPos] = coreNewLine[i];
-                    charPos++;
+                    _charBuffer[_charPos++] = CoreNewLine[i];
                 }
             }
 
-            if (autoFlush)
+            if (AutoFlush)
```

Comment:
Same as https://github.com/dotnet/runtime/pull/44495#discussion_r520939136
```suggestion
            if (_autoFlush)
```

Reply:
Thanks.